src/lib.rs

```diff
@@ -1956,6 +1956,15 @@ pub fn digraph_all_pairs_dijkstra_path_lengths(
     graph: &digraph::PyDiGraph,
     edge_cost_fn: PyObject,
 ) -> PyResult<PyObject> {
+    if graph.node_count() == 0 {
+        return Ok(PyDict::new(py).into());
```

Comment:
Is this done just to return an empty dict if is an empty graph?

Reply:
Yep, exactly. Just fast path returning an empty dict for empty graphs.